api/server/v1/marshaler.go

```diff
@@ -16,18 +16,61 @@ package api
 
 import (
 	"encoding/json"
+	"io"
+	"net/url"
 	"time"
 
+	"github.com/ajg/form"
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 	jsoniter "github.com/json-iterator/go"
 	spb "google.golang.org/genproto/googleapis/rpc/status"
 	"google.golang.org/protobuf/proto"
 )
 
+const (
+	refreshToken = "refresh_token"
+)
+
+type CustomDecoder struct {
+	formDecoder *form.Decoder
+	jsonDecoder *json.Decoder
+	reader      io.Reader
+}
+
+func (f CustomDecoder) Decode(dst interface{}) error {
+	switch dst.(type) {
+	case *GetAccessTokenRequest:
+		{
+			byteArr, err := io.ReadAll(f.reader)
+			if err != nil {
+				return err
+			}
+			return unmarshalInternal(byteArr, dst)
+		}
+	}
+	return f.jsonDecoder.Decode(dst)
+}
+
 // CustomMarshaler is a marshaler to customize the response. Currently, it is only used to marshal custom error message
 // otherwise it just uses the inbuilt mux marshaller.
 type CustomMarshaler struct {
-	*runtime.JSONBuiltin
+	JsonBuiltin *runtime.JSONBuiltin
```

Comment:
We can keep embedded version or we should rename the member to JSONBuiltin otherwise linters would complain. It expectes JSON to be capitalized.

Reply:
updated the variable name and keeping the variable here to be clear explicitly on how the methods are implemented.